internal/transactionsubmission/engine/signing/distribution_account_resolver.go

```diff
@@ -59,34 +66,56 @@ var _ DistributionAccountResolver = (*DistributionAccountResolverImpl)(nil)
 type DistributionAccountResolverImpl struct {
 	tenantManager                 tenant.ManagerInterface
 	hostDistributionAccountPubKey string
+	circleConfigModel             circle.ClientConfigModelInterface
 }
 
 // DistributionAccount returns the tenant's distribution account stored in the database.
 func (r *DistributionAccountResolverImpl) DistributionAccount(ctx context.Context, tenantID string) (schema.TransactionAccount, error) {
-	return r.getDistributionAccount(r.tenantManager.GetTenantByID(ctx, tenantID))
+	tnt, err := r.tenantManager.GetTenantByID(ctx, tenantID)
+	if err != nil {
+		return schema.TransactionAccount{}, fmt.Errorf("getting tenant: %w", err)
+	}
+	tenant.SaveTenantInContext(ctx, tnt)
+	return r.getDistributionAccount(ctx, tnt)
 }
 
 // DistributionAccountFromContext returns the tenant's distribution account from the tenant object stored in the context
 // provided.
 func (r *DistributionAccountResolverImpl) DistributionAccountFromContext(ctx context.Context) (schema.TransactionAccount, error) {
-	return r.getDistributionAccount(tenant.GetTenantFromContext(ctx))
-}
-
-// getDistributionAccount extracts the distribution account from the tenant if it exists.
-func (r *DistributionAccountResolverImpl) getDistributionAccount(tnt *tenant.Tenant, err error) (schema.TransactionAccount, error) {
+	tnt, err := tenant.GetTenantFromContext(ctx)
 	if err != nil {
 		return schema.TransactionAccount{}, fmt.Errorf("getting tenant: %w", err)
 	}
+	return r.getDistributionAccount(ctx, tnt)
+}
 
-	if tnt.DistributionAccountAddress == nil {
-		return schema.TransactionAccount{}, ErrDistributionAccountIsEmpty
-	}
+// getDistributionAccount extracts the distribution account from the tenant if it exists.
+func (r *DistributionAccountResolverImpl) getDistributionAccount(ctx context.Context, tnt *tenant.Tenant) (schema.TransactionAccount, error) {
+	if tnt.DistributionAccountType == schema.DistributionAccountCircleDBVault {
+		// 1. Circle Account
+		cc, circleErr := r.circleConfigModel.Get(ctx)
+		if circleErr != nil {
+			return schema.TransactionAccount{}, fmt.Errorf("getting circle client config: %w", circleErr)
+		}
+
+		return schema.TransactionAccount{
+			CircleWalletID: *cc.WalletID,
+			Type:           schema.DistributionAccountCircleDBVault,
+			Status:         schema.AccountStatusActive,
```

Comment:
👋  We should forward the status stored in the tenant, rather than hardcoding it to active.

```suggestion
			Status:         tnt.DistributionAccountStatus,
```

NOTE: in the future, we'll eventually need to implement a mechanism to update the status in the tenant from Active->Pending user when we get 401s from Circle.

Reply:
I addressed it in https://github.com/stellar/stellar-disbursement-platform-backend/pull/328/commits/e61716ec1a7dbdf62089e75a31a3dd3b892c4de9.